hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/viewfs/ViewFileSystemBaseTest.java

```diff
@@ -1102,6 +1104,131 @@ public void testTrashRoot() throws IOException {
     Assert.assertTrue("", fsView.getTrashRoots(true).size() > 0);
   }
 
+  /**
```

Comment:
Please add a test where the child file system returns a root inside of the mount point, such as what will happen when there is an encryption zone root. To do so, you'll need to define a mock/stub fs.

Reply:
Hi @omalley, 

I believe Case 5 is what you are looking for. 😃

Reply:
Actually, it isn't. I was more concerned with what would happen with an encryption zone or snapshot root. You'd need to make a mock child file system that implements a custom getTrashRoot(path).